iothub_client/samples/iothub_client_device_cbor_sample/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+if(NOT ${use_mqtt})
+    message(FATAL_ERROR "iothub_client_device_cbor_sample being generated without MQTT support")
+endif()
+
+compileAsC99()
+
+IF(WIN32)
+    #windows needs this define
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    add_definitions(-DGB_MEASURE_MEMORY_FOR_THIS -DGB_DEBUG_ALLOC)
+ENDIF(WIN32)
+
+if(MSVC)
+  set(CMAKE_EXE_LINKER_FLAGS
+      "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:LIBCMT.lib") # To compile with intel/tinycbor
+endif()
+
+# Sample Executable
+set(iothub_client_device_cbor_sample_c_files
+    iothub_client_device_cbor_sample.c
+)
+
+if(${use_sample_trusted_cert})
+    add_definitions(-DSET_TRUSTED_CERT_IN_SAMPLES)
+    include_directories(${PROJECT_SOURCE_DIR}/certs)
+    set(iothub_client_device_cbor_sample_c_files ${iothub_client_device_cbor_sample_c_files}
+        ${PROJECT_SOURCE_DIR}/certs/certs.c)
+endif()
+
+add_executable(iothub_client_device_cbor_sample
+    ${iothub_client_device_cbor_sample_c_files}
+)
```

Comment:
yes.  The other sample cmakes take the other approach and I wasn't planning to deviate from the existing pattern. however, if your suggested approach is a more modern way of doing the cmake, i'm fine with these changes.

Reply:
gotcha understood. Yes this would be the preferred way to do it. I would rather start transitioning at least new additions to the more desired way of doing things instead of accumulating more technical debt that would be updated later.

Reply:
Cool.  I'll make the changes.  I'm trying to take a similar approach on the samples -- they all need updating, so starting with CBOR.